helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -692,8 +701,8 @@ private void handleEvent(ClusterEvent event, BaseControllerDataProvider dataProv
               forceRebalance(manager, ClusterEventType.RetryRebalance);
             } else {
               _asyncTasksThreadPool
-                  .schedule(new RebalanceTask(manager, ClusterEventType.RetryRebalance), delay,
-                      TimeUnit.MILLISECONDS);
+                  .schedule(new RebalanceTask(manager, ClusterEventType.RetryRebalance, -1, false),
```

Comment:
Nit: -1 -> constant (as mentioned above) for readability?

Reply:
Let me do this after the merge, tracking in issue #640